design-guidelines/form-ux.md

```diff
@@ -0,0 +1,189 @@
+
+### Field order
+
+Order fields by their relative level of importance. The inputs that are most critical for the user to complete their task should come first.
+
+### Field grouping
+
+Fields that are closely related should be visually grouped together.
+
+For more information on grouped fields, see the Storybook documentation for the `FieldList` and `FieldGroup` components.
+
+### Form actions
+
+A form should contain at least two buttons: one to submit the form, and one to navigate away from the form without submitting any of the input.
+
+These buttons should be highly visible, and easy for the user to press.
+
+### Multi-part forms
+
+Forms should be kept as succinct as possible, but some forms require a lot of data input. Break complex forms into multiple forms when it will help the user focus on what data they're inputting.
+
+#### Tabbed form pattern
+
+Break a form into different tabs when a form is used to configure an object with so many data points that there is no way to keep the fields on one screen without overwhelming the user. For example: a form that is used to deploy an addon with many configuration options.
+
+Keep in mind that breaking a form into separate tabs forces the user to navigate to find a specific field. Only use this pattern if tabbed sections make it easier to find a specific field.
+
+#### Progressive disclosure pattern
+
+If a complex form can be broken into a linear series of steps, sections can be progressively shown as the user inputs their data.
+
+A progressive disclosure pattern is also useful when the value of one field will affect subsequent fields. For example: selecting a cloud provider before showing the fields for authorization credentials.
+
+## Kinds of inputs
+
+### Text inputs
+
+Use text inputs when the field value is expected to be free-form text, and we can't provide a set of possible values for a user to choose from. For example: setting a cluster's display name.
+
+For more detailed information on text inputs, see the Storybook documentation for the `TextInput` and `Textarea` components.
+
+### Choice inputs
+
+Use a choice input when a field has a known set of possible values. For example: selecting which version of Kubernetes should be used when creating a cluster.
+
+Some examples of choice inputs include: typeahead dropdowns, `<select>` dropdowns, checkboxes, file upload fields, and segmented controls.
+
+For more detailed information on choice inputs, see the respective Storybook documentation for the respective choice input components (e.g.: `CheckboxInput`, `SegmentedControl`, `SelectInput`, `ToggleBox`, etc)
+
+## Field parts
+
+A field can accept data in a variety of ways (e.g.: typing, dropdown selection, etc.), but they are generally made up of the same parts.
+
+![Input field diagram](https://raw.githubusercontent.com/dcos-labs/ui-kit/master/design-guidelines/img/form-ux-patterns/inputFieldDiagram.png)
+
+### Label
```

Comment:
what happens when we click the label? in dcos-ui we have a multitude of places where clicking a seemingly corresponding label does nothing, while it can focus the associated input for you. i feel like we could commit to always set those `for` and `id` props here. :)

Reply:
Sadly, you're correct about the fields in dcos-ui. However, ui-kit form components are set up to take care of this :)